sync/sync-impl/src/main/java/com/duckduckgo/sync/impl/auth/DeviceAuthenticator.kt

```diff
@@ -92,4 +103,45 @@ class RealDeviceAuthenticator @Inject constructor(
             onResult(AuthResult.Success)
         }
     }
+
+    @SuppressLint("InlinedApi", "DEPRECATION")
```

Comment:
Duplicating this between autofill and sync (and whichever other feature needs all of this) isn't a great strategy long term. It's _ok_ for now probably, but we should start identifying which parts are truly common and pulling them out for re-use vs duplicating.